chain/jsonrpc/src/lib.rs

```diff
@@ -283,16 +287,68 @@ impl JsonRpcHandler {
         .map_err(|_| timeout_err())?
     }
 
+    async fn send_tx(
+        &self,
+        tx: SignedTransaction,
+        check_only: bool,
+    ) -> Result<NetworkClientResponses, RpcError> {
+        Ok(self
+            .client_addr
+            .send(NetworkClientMessages::Transaction {
+                transaction: tx,
+                is_forwarded: false,
+                check_only,
+            })
+            .map_err(|err| RpcError::server_error(Some(ServerError::from(err))))
+            .await?)
+    }
+
+    async fn send_tx_sync(&self, params: Option<Value>) -> Result<Value, RpcError> {
+        self.send_or_check_tx(params, false).await
+    }
+
+    async fn check_tx(&self, params: Option<Value>) -> Result<Value, RpcError> {
+        self.send_or_check_tx(params, true).await
+    }
+
+    async fn send_or_check_tx(
+        &self,
+        params: Option<Value>,
+        check_only: bool,
+    ) -> Result<Value, RpcError> {
+        let tx = parse_tx(params)?;
+        let tx_hash = (&tx.get_hash()).to_base();
```

Comment:
We migrate from the positional arguments to the RPC API. Define `RpcTransactionStatusCheckRequest` in `primitives/rpc.rs` and parse it here as:

```rust
let params: RpcTransactionStatusCheckRequest = parse_params(params)?;
```

The new style API is already implemented for `query`, `EXPERIMENTAL_*` methods, so you can take inspiration from there.

Reply:
Well transaction is special because it is serialized in borsh. That is why we have `parse_tx`. I can wrap it in an object but I don't see the point of doing so, as you still have to use a special routing to deserialize it.